connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/util/Cleaner.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.client.util
+
+import java.lang.ref.{ReferenceQueue, WeakReference}
+import java.util.Collections
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.mutable
+import scala.util.control.NonFatal
+
+/**
+ * Helper class for cleaning up an object's resources after the object itself has been garbage
+ * collected.
+ *
+ * When we move to Java 9+ we should replace this class by [[java.lang.ref.Cleaner]].
```

Comment:
Apache Spark supports Java 11 already. 
May I ask if this mean `connect` module only support Java 8?

Reply:
This is the Scala client. The java version requirement can be different from the Spark Connect Server or the Spark.

Reply:
Oh, got it. So, we still can claim Python Connect Client works fine with Spark Connect Server with Java 11+, right?

Reply:
Yes, you can say so. The Spark Connect Server must be compatible with Spark. The clients can be remote thus can have different java version requirements.